doc/src/manual/modules.md

```diff
@@ -1,92 +1,203 @@
 # [Modules](@id modules)
 
-Modules in Julia are separate variable workspaces, i.e. they introduce a new global scope. They
-are delimited syntactically, inside `module Name ... end`. Modules allow you to create top-level
-definitions (aka global variables) without worrying about name conflicts when your code is used
-together with somebody else's. Within a module, you can control which names from other modules
-are visible (via importing), and specify which of your names are intended to be public (via exporting).
+Modules in Julia help organize code into coherent units. They are delimited syntactically inside
+`module NameOfModule ... end`, and have the following features:
 
-The following example demonstrates the major features of modules. It is not meant to be run, but
-is shown for illustrative purposes:
+1. Modules are separate namespaces, each introducing a new global scope. This is useful, because it
+   allows the same name to be used for different functions or global variables without conflict, as long as they are in separate modules.
 
-```julia
-module MyModule
-using Lib
+2. Modules have facilities for detailed namespace management: each defines a set of symbols it
+   `export`s, and can import symbols from other modules with `using` and `import` (we explain these below).
+
+3. Modules can be precompiled for faster loading, and contain code for runtime initialization.
 
-using BigLib: thing1, thing2, thing3 as t3
+Typically, in larger Julia packages you will see module code organized into files, eg
+
+```julia
+module SomeModule
 
-import Base.show, Base.print as pr
+# export, using, import statements are usually here; we discuss these below
 
-export MyType, foo
+include("file1.jl")
+include("file2.jl")
 
-struct MyType
-    x
 end
+```
+
+Files and file names are mostly unrelated to modules; modules are associated only with module
+expressions. One can have multiple files per module, and multiple modules per file, `include`
+behaves as if the contents of the source file were evaluated in its place. In this chapter, we use
+short and simplified examples, so we won't use `include`.
+
+The recommended style is not to indent the body of the module, since that would typically lead to
+whole files being indented. Also, it is common to use `MixedCase` for module names (just like
+types), and use the plural form if applicable, especially if the module contains a similarly named
```

Comment:
In https://github.com/JuliaLang/julia/blob/4562ecef9939a8f132c706213fd628f5d0978c55/doc/src/manual/variables.md#stylistic-conventions this is referred to as "upper camel case"

Reply:
Done, thanks!